tools/codegen/core/gen_experiments.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+
+# Copyright 2022 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import collections
+import ctypes
+import datetime
+import json
+import math
+import os
+import re
+import sys
+
+import yaml
+
+with open('src/core/lib/experiments/experiments.yaml') as f:
+    attrs = yaml.load(f.read(), Loader=yaml.FullLoader)
+
+error = False
+today = datetime.date.today()
+two_quarters_from_now = today + datetime.timedelta(days=180)
+for attr in attrs:
+    if 'name' not in attr:
+        print("experiment with no name: %r" % attr)
+        error = True
+        continue
```

Comment:
should there be a continue for each if statement?

Reply:
no there needn't be - but we need to ensure there's a name so we can print later diagnostics